port/unix/omrsysinfo.c

```diff
@@ -3317,31 +3345,41 @@ getCgroupNameForSubsystem(struct OMRPortLibrary *portLibrary, OMRCgroupEntry *cg
 static int32_t
 addCgroupEntry(struct OMRPortLibrary *portLibrary, OMRCgroupEntry **cgEntryList, int32_t hierId, const char *subsystem, const char *cgroupName)
 {
-	OMRCgroupEntry *cgEntry = NULL;
-	int32_t cgEntrySize = sizeof(OMRCgroupEntry) + strlen(subsystem) + 1 + strlen(cgroupName) + 1;
 	int32_t rc = 0;
+	int32_t i = 0;
 
-	cgEntry = portLibrary->mem_allocate_memory(portLibrary, cgEntrySize, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
-	if (NULL == cgEntry) {
-		rc = portLibrary->error_set_last_error_with_message(portLibrary, OMRPORT_ERROR_SYSINFO_MEMORY_ALLOC_FAILED, "memory allocation for cgroup entry failed");
-		goto _end;
-	}
-	cgEntry->hierarchyId = hierId;
-	cgEntry->subsystem = (char *)(cgEntry + 1);
-	strcpy(cgEntry->subsystem, subsystem);
-	cgEntry->cgroup = cgEntry->subsystem + strlen(subsystem) + 1;
-	strcpy(cgEntry->cgroup, cgroupName);
+	for (i = 0; i < sizeof(supportedSubsystems) / sizeof(supportedSubsystems[0]); i++) {
+		if (J9_ARE_NO_BITS_SET(PPG_cgroupSubsystemsAvailable, supportedSubsystems[i].flag)
```

Comment:
Is it possible for two threads to be executing this method at once?  If it's not, then the function should document what lock needs to be held when calling it.  Otherwise, it shouldn't be reading / modifying global state (`PPG_cgroupSubsystemsAvailable`) without a lock.

Reply:
I have updated `addCgroupEntry()` to avoid using PPG_cgroupSubsystemsAvailable.
Available subsystems are now determined by the `readCgroupFile()` and returned back via one of its parameters.